update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,750 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// UpdateStatus is the API about in-progress updates, kept populated by Update Status Controller by
+// aggregating and summarizing UpdateInsights produced by update informers
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=updatestatuses,scope=Namespaced
+// +openshift:api-approved.openshift.io=TODO
+// +openshift:file-pattern=cvoRunLevel=0000_00,operatorName=cluster-version-operator,operatorOrdering=02
+// +openshift:enable:FeatureGate=UpgradeStatus
+// +kubebuilder:metadata:annotations="description=Provides health and status information about OpenShift cluster updates."
+// +kubebuilder:metadata:annotations="displayName=UpdateStatuses"
+type UpdateStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec UpdateStatusSpec `json:"spec"`
+	// +optional
+	Status UpdateStatusStatus `json:"status,omitempty"`
+}
+
+// UpdateStatusSpec is empty for now, can possibly hold configuration for Update Status Controller in the future
+type UpdateStatusSpec struct {
+}
+
+// +k8s:deepcopy-gen=true
+
+// UpdateStatusStatus is the API about in-progress updates, kept populated by Update Status Controller by
+// aggregating and summarizing UpdateInsights produced by update informers
+type UpdateStatusStatus struct {
+	// controlPlane contains a summary and insights related to the control plane update
+	ControlPlane ControlPlaneUpdateStatus `json:"controlPlane"`
+
+	// workerPools contains summaries and insights related to the worker pools update
+	WorkerPools []PoolUpdateStatus `json:"workerPools"`
+
+	// conditions provide details about Update Status Controller operational matters
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
nit: [upstream Godocs][1] suggest this get some `patchMergeKey`, `patchStrategy`, and `protobuf` stuff too.  I dunno if we need to do that or not; grepping around seems like there's lots of diversity within openshift/api's consumption of `metav1.Condition`.

[1]: https://github.com/kubernetes/apimachinery/blob/v0.31.1/pkg/apis/meta/v1/types.go#L1551-L1554

Reply:
I don't see `protobuf` used on the types I used as reference so I did not add that. For the other I found https://github.com/kubernetes-sigs/gateway-api/issues/342 which says:

> +patchMergeKey and +patchStrategy are for guiding strategic merge, which is not supported on CRDs, so those annotations do nothing. It's sufficient to use just +listType and +listMapKey.

So I'm not adding these for now.